src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -277,8 +289,52 @@ private static Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibil
                     }
                 }
             }
+        }
+
+        private static void AddNamesFromExistingOverloads(CSharpSyntaxContext context, SemanticModel semanticModel, ArrayBuilder<(string name, SymbolKind kind)> result, CancellationToken cancellationToken)
+        {
+            var namedType = semanticModel.GetEnclosingNamedType(context.Position, cancellationToken);
+            if (namedType is null)
+                return;
+
+            var parameterSyntax = context.LeftToken.GetAncestor(n => n.IsKind(SyntaxKind.Parameter)) as ParameterSyntax;
+            if (parameterSyntax is not { Type: { } parameterType, Parent: { Parent: BaseMethodDeclarationSyntax baseMethod } })
```

Comment:
```suggestion
            if (parameterSyntax is not { Type: { } parameterType, Parent.Parent: BaseMethodDeclarationSyntax baseMethod })
```

Reply:
also, we don't want this in the optional initializer value of a parameter either.

Reply:
> also, we don't want this in the optional initializer value of a parameter either.

I'm not sure what you mean.

```csharp
// I think this is not handled by declaration name completion at all?
public void M(string s = $$)
{
}
```

Reply:
basically, i'm saying that this test doesn't guarantee that (even if some other check does).  So it would be good to have a test here for those cases, just to show there's no issue.

Reply:
@CyrusNajmabadi Yeah sure adding the test is good for coverage anyway even if already handled. I was trying to make sure I'm properly understanding the scenario you're referring to.

Reply:
yup.  that's the scenario.  the core issue i have with this code is that all it does is check if it's contained within a parameter.

Now, it sounds like there's other code that ensures that we're typing the parameter name.  If so, that's great.  I just want tests to ensure that works properly.  thanks!